[EFA] Improvements on messaging used in the EFA Validator

EddyMM opened the pull request:
When creating or updating a cluster that has instance-types that support EFA (but EFA is not enabled in the configuration), the user should get a brief but informative validation message about the benefits of enabling EFA


### Description of changes
* To communicate the benefits of EFA more clearly, improvements have been made on the messaging used in the EFA Validator when creating or updating EFA supported instance-types

```
"validationMessages": [
    {
      "level": "WARNING",
      "type": "EfaValidator",
      "message": "The EC2 instance selected (c5n.9xlarge) supports enhanced networking capabilities using Elastic Fabric Adapter (EFA). EFA enables you to run applications requiring high levels of inter-node communications at scale on AWS at no additional charge. You can update the cluster's configuration to enable EFA (https://docs.aws.amazon.com/parallelcluster/latest/ug/efa-v3.html)"
    }
  ]
```

### Tests
* Unit tests

### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

enrico-usai commented on cli/src/pcluster/validators/cluster_validators.py:
```diff
@@ -309,8 +309,10 @@ def _validate(self, instance_type, efa_enabled, gdr_support):
-                f"To get results faster with the instance type '{instance_type}' at no additional charge, enable the "
-                "Elastic Fabric Adapter (https://docs.aws.amazon.com/parallelcluster/latest/ug/efa.html)",
+                f"The EC2 instance selected ({instance_type}) supports enhanced networking capabilities using "
+                f"Elastic Fabric Adapter (EFA). EFA enables you to run applications requiring high levels of "
```
remove `f`  from strings without variables on it.

enrico-usai replied:
You can just keep it in the first line.

EddyMM committed:
[EFA] Improvements on messaging used in the cluster validator for EFA supported instance-types

When creating or updating a cluster that has instance-types that support EFA (but EFA is not enabled in the configuration), the user should get a brief but informative validation message about the benefits of enabling EFA
